### DISCUSSION TITLE
Calculation of forces with PyMBD

### DISCUSSION THREAD

**Discussion by mgori88** [libmbd/libmbd#45] [Q&A] [answered]
Dear developers,

thank you for your efforts in implementing such a useful tool.

I am trying to calculate forces with pymbd, but I am encountering some difficulties. Looking at LibMBD procedures, 
the derivatives of polarizability and C6 coefficient with nuclei position would be required. Is it possible to calculate MBD forces without having access to this information?

Many thanks for considering my request.

**Comment by jhrmnn** [comment, marked as answer]
Hi, and sorry for the delayed response, this is not the norm.

Indeed, to get the exact forces, one needs to supply the partial derivatives of the vdW parameters w.r.t. oscillator coordinates. But often, in practice, this term is almost negligible. I have never encountered a case where it would disturb a geometry optimization, for example.

**Comment by mgori88** [comment]
Dear Jan,

Thank you very much answer, and sorry for my very late reply.

Under the suggestion of prof. Tkatchenko, I would plot the charge density
distribution of QDOs: do you know if this is possible with
LibMBD?

All the best,
Matteo

Il giorno lun 21 nov 2022 alle ore 22:21 Jan Hermann <
***@***.***> ha scritto:

**Reply by jhrmnn** [reply]
Hi, I know that this is most likely too late, but just for reference, there’s now an example in the documentation for this, https://libmbd.nablapsi.science/pymbd/densities.html
